src/fec_rs_nf4.h

```diff
@@ -67,19 +66,26 @@ class RsNf4 : public FecCode<T> {
         // n_data)
         n = sub_field->get_code_len_high_compo(n_parities + n_data);
 
-        // compute root of order n-1 such as r^(n-1) mod q == 1
-        r = sub_field->get_nth_root(n);
+        // compute root of order n-1 such as r^(n-1) mod q == (1, ..,1)
+        this->r = ngff4->get_nth_root(this->n);
 
         // std::cerr << "n=" << n << "\n";
         // std::cerr << "r=" << r << "\n";
 
         int m = arith::get_smallest_power_of_2<int>(n_data);
-        this->fft = new fft::Radix2<T>(ngff4, n, m);
+        this->fft = new fft::Radix2<T>(ngff4, this->n, m);
+
+        // vector stores r^{-i} for i = 0, ... , k
+        T inv_r = ngff4->inv(this->r);
+        this->inv_r_powers = std::unique_ptr<vec::Vector<T>>(
+            new vec::Vector<T>(ngff4, this->n_data + 1));
+        for (int i = 0; i <= this->n_data; i++)
+            this->inv_r_powers->set(i, ngff4->exp(inv_r, i));
     }
 
     ~RsNf4()
     {
-        delete fft;
+        delete this->fft;
```

Comment:
`fft` was declared as a member of FEC::NF4: https://github.com/scality/nttec/pull/203/commits/e9613f8ec7caa66b2987978cf8f30cb4a570547b#diff-153495ed6caf85a68307a8491242e39cL145
I removed it since it's declared in the base class.
A separated commit will define fft and gf as a unique_ptr.